### DISCUSSION TITLE
Alternate logic - modelling connected physical systems

### DISCUSSION THREAD

**Discussion by spettitt9095** [Akascape/TkNodeSystem#4] [Ideas]
I am trying to build an application where users can quickly drop in nodes that represent real world equipment with input and output sockets, and connect them up.

As opposed to the current framework, I would need each node to have no more than one input, and multiple outputs (up to about 30). This would model power distribution - where there is an electrical supply (a value node), which forks off to one or more distributors. The power would never merge, only split.

All I really need is for each node to know the ID of the source socket and source node that supplied it.

Do you think it would be possible to modify the logic such that there is only one input and multiple outputs, rather than the other way around? If so, do you have any suggestions where I should start?

Thank you

**Comment by Akascape** [comment]
@spettitt9095 
That's possible, I will try to implement multiple output sockets in the operation and value node. The logic will be to implement separate functions for each socket. Currently we only have one function which takes inputs and returns one output.

**Comment by spettitt9095** [comment]
oh thank you! I wasn't meaning to ask you to do it, but if it's something you'd like to look at then thanks!

I have also been considering whether each NodeSocket object could have a text ID shown next to the oval, such that the user can see 'Output A', 'Output B' etc. I'm still quite new to Python but I can have a go at that unless it's something you'd like to add as well.

**Reply by Akascape** [reply]
@spettitt9095 
> I have also been considering whether each NodeSocket object could have a text ID shown next to the oval, such that the user can see 'Output A', 'Output B' etc. 

We have hover tooltips for each socket which can display the details.

![242548423-413a546f-e526-484d-add8-df079122a372.jpg](https://github.com/user-attachments/assets/c0b40e4f-67cd-4b53-97c1-3e6a57a3ab3c)

We can also change color for each socket.

**Reply by iwr-redmond** [reply]
I agree with @spettitt9095, this would be very helpful. Put together with a fix for #3, the package would give us more than we can say grace over!
